### ISSUE TITLE
Questions regarding output generated

### ISSUE THREAD

**Issue by chsheth** [google/lightweight_mmm#66]
I have a dataset with 16 media channel spend and 1 extra features. This data is over 209 weeks and is sparse. My dependent variable is conversion. Additionally the difference in media channel spend is in orders of magnitude (1000 vs. 1000000). 

When I use utils.dataframe_to_jax() to convert my pandas dataframe to jax, 'geo_feature' is a required argument -- not sure how this works if the data is at national level. I created a 'geo' variable in my dataframe that has all 1s, and got this function to work, but not sure if this is how it should be.

I have the notebook setup exactly like the simpl_demo notebook (scalars, priors etc.). When I run mmm.fit, I get r_hat that is in the 100,000 range.  The plotting function doesnt work and roi_hat from mmm_get_posterior_metrics is all 'nan'.

I need some help troubleshooting -- cannot think of what to change/try to get some output. Any suggestions?

Thanks.

**Comment by pabloduque0** [comment]
Hello @chsheth !

If you are using the national model and coming from a `pd.DataFrame` you can just take the values of the dataframe directly, `utils.dataframe_to_jax()` is only needed for geo models.

Something like the following should suffice:
```
dataframe = pd.DataFrame(YOUR DATA HERE)
media = jnp.array(dataframe[media_columns].values)
extra_features = jnp.array(dataframe[extra_feature_columns].values)
```

Yes, trying to fit a geo model with national data and a "faked" geo column might lead to weird outputs. Hopefully trying with the correct model helps!

**Comment by chsheth** [comment]
Hi @pabloduque,

Thanks again for your support.

I have created the dataframe similarly to how you have proposed. Here is my
code -

media_data = weekly_df[['x1', 'x2', 'x3',]]

extra_features = weekly_df['unsol']
date_feature = weekly_df['date'] #not used
target = weekly_df['convs']

# costs to make ROI realistic.
costs = jnp.array(media_data.sum(axis=0))

media_data = jnp.array(media_data)
extra_features = jnp.reshape(jnp.array(extra_features), (-1,1))
target = jnp.reshape(jnp.array(target), (-1, 1))

------- Media features - shape (104, 16) Extra features - shape (104, 1)
Target - shape (104, 1) Costs - shape (16,)
As you can see above, I have similar shapes to what the simulated data has.
The code now runs and I get r_hat close to 1.

However, I had the following questions -
1. The model output has a bunch of coefficients -- is there anywhere I
could read what they mean and interpret them within the context?
2. The following plot commands don't work. I dont know if it is an arviz
issue or some data/shape issue.
---- plot.plot_model_fit(mmm, target_scaler=target_scaler) &
----
plot.plot_out_of_sample_model_fit(out_of_sample_predictions=new_predictions,

out_of_sample_target=target_scaler.transform(target[split_point:]))
3. The optimization indicates that there are bounds of 20% -- I dont see
where these can be specified for each channel.
Thanks again for the help.

**Comment by pabloduque0** [comment]
Sorry for the late response, I missed this one.

1. Yes the [model documentation](https://lightweight-mmm.readthedocs.io/en/latest/models.html) and [custom priors documentation](https://lightweight-mmm.readthedocs.io/en/latest/custom_priors.html) should shed some light about all parameters in the model.
2.  For specifying bounds per channel you need to pass an array, eg. if you have three chanenls your bounds could be jnp.array([0.25, 0.3, 0.15]) and the parameters are: `bounds_lower_pct`and `bounds_upper_pct`.

**Comment by chsheth** [comment]
Thanks @pabloduque0
